packages/api-codec/src/codec/Struct.ts

```diff
@@ -6,59 +6,83 @@ import u8aConcat from '@polkadot/util/u8a/concat';
 
 import Base from './Base';
 
-// A Struct defines an Object with key/values - where the values are CodecBase values. It removes
+// A Struct defines an Object with key/values - where the values are Base<T> values. It removes
 // a lot of repetition from the actual coding, define a structure type, pass it the key/Base<T>
 // values in the constructor and it manages the decoding. It is important that the constructor
 // values matches 100% to the order in th Rust code, i.e. don't go crazy and make it alphabetical,
 // it needs to decoded in the specific defined order.
-//
-// TODO:
-//   - Check the constructor, something is really, really wrong with the way the defs are used
 export default class Struct <
   S = { [index: string]: { new(value?: any): Base } },
   T = { [K in keyof S]: Base },
-  V = { [K in keyof S]: any }
+  V = { [K in keyof S]: any },
+  E = { [K in keyof S]: string }
```

Comment:
Can you provide an example in the comments on what each type STVE represents?

Reply:
Done, pushing.